### PR TITLE
Add pytest Solaris 11.4 IPS packaging tests

### PR DESCRIPTION
**Pull request by dmurphy18**
### What does this PR do?
Adds pytest based tests for Solaris 11.4 IPS packaging, install / remove / get_fmri and their different flavors.
This is part of the preparation for introduction of Python 3 Tiamat based native minion support for Solaris 11.4 (Intel - completed, Sparc - under development)

### What issues does this PR fix or reference?
Fixes:
https://github.com/saltstack/salt/issues/59306
https://gitlab.com/saltstack/open/salt-pack-extras/-/issues/184

### Previous Behavior
Used old unit non-pytest based IPS packaging for Solaris 11.3 (which is EOL'd and version format was different), and were Python 2 based.

### New Behavior
Adds new pytest based tests for IPS packaging and allows for package version format change in Solaris 11.4 IPS packages

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [X] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by dmurphy18** [comment]
re-run macos

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: tests/pytests/unit/modules/test_solarisipspkg.py [hunk 1]
```diff
@@ -0,0 +1,437 @@
+import logging
+
+import salt.modules.solarisipspkg as solarisipspkg
+from tests.support.mock import MagicMock, call, patch
+
+try:
+    import pytest
+except ImportError:
+    pytest = None
+
```
[the remark this answers is not in the document]
Comment (dmurphy18) [reply tests/pytests/unit/modules/test_solarisipspkg.py]: Made the change, re-running tests
That is what I get for looking at test_yumpkg.py as an example.  should remove the same code from there as part of clean up at some point

**Comment by dmurphy18** [comment]
re-run all

**Comment by dmurphy18** [comment]
re-run all

**Review by s0undt3ch — APPROVED** [review]

**Comment by dmurphy18** [comment]
re-run all

**Review by Ch3LL — APPROVED** [review]

**Review by krionbsd — APPROVED** [review]

**Review by twangboy — APPROVED** [review]

**Pull request merged**